service/history/shard/context.go

```diff
@@ -863,6 +863,9 @@ func (s *contextImpl) PreviousShardOwnerWasDifferent() bool {
 }
 
 func (s *contextImpl) GetEventsCache() events.Cache {
+	if s.config.EventsCacheGlobalEnable() {
+		return s.GetEventCache()
+	}
```

Comment:
I want to make sure my understanding is correct here. Is your goal here to make the dynamic config `EventsCacheGlobalEnable` truly dynamic (meaning it should take effect without service restart)? If so, then one small problem here is that if an execution is cached, then it'll keep using the old cache until it's reloaded.